hack/charts/cluster-api-operator/templates/deployment.yaml

```diff
@@ -100,9 +97,6 @@ spec:
         - containerPort: 9443
           name: webhook-server
           protocol: TCP
-        - containerPort: {{ ( split ":" $.Values.metricsBindAddr)._1 | int }}
```

Comment:
This should probably be replaced with .Values.diagnosticsAddress, that address is now not exposed at all

Also in that case add a condition around the containerPort because if diagnosticsAddress is set to '" by a chart user, containerPort would be set to 0 which is invalid. This is currently the case with metricsBindAddr

Reply:
I added the diagnostics address port with improved validation if it's set and condition parsing only if it contains the ':' char.